src/SfxWeb/src/app/views/cluster/commands/commands.component.ts

```diff
@@ -35,12 +35,84 @@ export class CommandsComponent extends BaseControllerDirective {
+      .map(x => { return x.x509FindValue });
+    let clientCommonNames = certificateProperties
+      .filter(x => x.name === "ClientCertificate" && x.x509FindType === "FindBySubjectName")
+      .map(x => { return x.x509FindValue });
+    let clusterThumbprints = certificateProperties
+      .filter(x => x.name === "ClusterCertificate" && x.x509FindType !== "FindBySubjectName")
+      .map(x => { return x.x509FindValue });
+    let clusterCommonNames = certificateProperties
+      .filter(x => x.name === "ClusterCertificate" && x.x509FindType === "FindBySubjectName")
+      .map(x => { return x.x509FindValue });
+    let serverThumbprints = certificateProperties
+      .filter(x => x.name === "ServerCertificate" && x.x509FindType !== "FindBySubjectName")
+      .map(x => { return x.x509FindValue });
+    let serverCommonNames = certificateProperties
+      .filter(x => x.name === "ServerCertificate" && x.x509FindType === "FindBySubjectName")
+      .map(x => { return x.x509FindValue });
+
+    const serverThumbprint = new PowershellCommandParameter("ServerCertThumbprint", CommandParamTypes.enum, {
+      required: true,
+      options: [...new Set(serverThumbprints)],
+      defaultValue: serverThumbprints[0]
+    });
+    const findValueThumbprint = new PowershellCommandParameter("FindValue", CommandParamTypes.enum, {
+      required: true,
+      options: [...new Set(clientThumbprints.concat(clusterThumbprints))],
+      defaultValue: clientThumbprints[0] ?? clusterThumbprints[0]
+    });
+
+    const serverCommonName = new PowershellCommandParameter("ServerCommonName", CommandParamTypes.enum, {
+      required: true,
+      options: [...new Set(serverCommonNames)],
+      defaultValue: serverCommonNames[0]
+    });
+    const findValueCommonName = new PowershellCommandParameter("FindValue", CommandParamTypes.enum, {
+      required: true,
+      options: [...new Set(clientCommonNames.concat(clusterCommonNames))],
+      defaultValue: clientCommonNames[0] ?? clusterCommonNames[0]
+    });
+
+    if (clientCommonNames.length > 0 || serverCommonNames.length > 0) {
+      const connectClusterCommon = new PowershellCommand(
+        'Creates a connection to a Service Fabric cluster using certificate common name.',
+        connectHelp,
+        CommandSafetyLevel.safe,
+        `Connect-ServiceFabricCluster ${authType}-StoreName My -FindType FindBySubjectName`,
+        [connectionEndpoint, storeLocation, serverCommonName, findValueCommonName, CommandFactory.GenTimeoutSecParam()]
+      );
+      this.commands.push(connectClusterCommon);
+    }
+
+    if (clientThumbprints.length > 0 || serverThumbprints.length > 0) {
+      const connectClusterThumbprint = new PowershellCommand(
+        'Creates a connection to a Service Fabric cluster using certificate thumbprint.',
+        connectHelp,
+        CommandSafetyLevel.safe,
+        `Connect-ServiceFabricCluster ${authType}-StoreName My -FindType FindByThumbprint`,
```

Comment:
When AAD is enabled, both `FindType` and `FindValue` are not needed, only Server Certificate is needed